### ISSUE TITLE
Support Vue 3

### ISSUE THREAD

**Issue by jods4** [TypeStrong/fork-ts-checker-webpack-plugin#384]
## Feature description
Vue 3 is now publicly available in alpha. 
Because of changes in its internal structure, tools like fork-ts-checker-webpack-plugin need to be updated (see below).

## Feature implementation
I'm using Vue 3 + Webpack + Typescript in my project.
This requires `ts-loader@6.2.1`, `vue-loader@16.0.0-alpha.3` and `@vue/compiler-sfc@3.0.0-alpha.4`.

When I try to use `vue: true` in `fork-ts-checker-webpack-plugin` I get the following error:
```
INTERNAL ERROR: When you use `vue` option, make sure to install `vue-template-compiler`.
```

I'm pretty sure this is because Vue 3 uses `@vue/compiler-sfc` to compile views instead of `vue-template-compiler` and the checker needs to be updated to use this package when it's available instead of the old one.

**Comment by johnnyreilly** [comment]
If you'd like to work on this that would be greatly appreciated. I'm not a Vue user and I believe none of the other maintainers are either. 

If someone would like to submit a PR that would be greatly appreciated. It's important that we maintain support for Vue 2 for now as well. Thanks!

**Comment by piotr-oles** [comment]
Did you try the following settings?
```
vue: {
  enabled: true,
  compiler: '@vue/compiler-sfc'
}
```

**Comment by jods4** [comment]
@piotr-oles I did not, I'll try tomorrow!
Should it work? If the packages share the same interfaces it might (and it would also be rather easy to fix out-of-the-box)

**Comment by jods4** [comment]
Quick update: it doesn't work.

Unfortunately the exported api has changed just a little bit.

I have not tested yet but changing this line:
https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/blob/master/src/VueProgram.ts#L321

to
```js
 const { descriptor: { script } } = parser.parse(content, {
      pad: 'space'
    });
```
might work.

**Comment by jods4** [comment]
PR available in #389

**Comment by jods4** [comment]
Tested with `5.0.0-alpha.9` it works great!
(after fiddling with the 5.0 config changes)

Note to anyone else finding this with Google: 
The correct compiler (Vue 2 vs 3) is not automatically detected, be sure to explicitly configure it in fork-ts-checker options:
```js
vue: {
  enabled: true,
  compiler: '@vue/compiler-sfc'
}
```
